File: RealEstate.Domain/Entities/Property.cs
namespace RealEstate.Domain.Entities;

public class Property
{
    public string Id { get; set; } = string.Empty;
    
    public string IdOwner { get; set; } = string.Empty;
    
    public string Name { get; set; } = string.Empty;
    
    public string Address { get; set; } = string.Empty;
    
    public decimal Price { get; set; }
    
    public string ImageUrl { get; set; } = string.Empty;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: real-estate-frontend/src/types/Property.ts
export interface Property {
  id: string;
  idOwner: string;
  name: string;
  address: string;
  price: number;
  imageUrl: string;
}

export interface CreatePropertyDto {
  idOwner: string;
  name: string;
  address: string;
  price: number;
  imageUrl: string;
}

export interface UpdatePropertyDto {
  name: string;
  address: string;
  price: number;
  imageUrl: string;
}

export interface PropertyFilter {
  name?: string;
  address?: string;
  minPrice?: number;
  maxPrice?: number;
  page?: number;
  pageSize?: number;
}
